test/e2e/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/notaryproject/notation/test/e2e
 
-go 1.21
+go 1.22
```

Comment:
I'm curious why the golang version for e2e test is not the same as the main one?

Reply:
Updated.